.circleci/config.yml

```diff
@@ -8,71 +8,106 @@ jobs:
       - checkout
       - run: git submodule sync
       - run: git submodule update --init
-      - restore_cache:
-          keys:
-            - cargo-v2-{{ checksum "Cargo.toml" }}-
-            - cargo-v2-
       - run: cargo fetch
-      - persist_to_workspace:
-          root: "."
-          paths:
-            - Cargo.lock
       - save_cache:
-          key: cargo-v2-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
+          key: cargo-cache-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
           paths:
             - /usr/local/cargo/registry
             - /usr/local/cargo/git
-  test:
+  rustfmt:
+    docker:
+      - image: rust:latest
+    working_directory: /mnt/crate
+    steps:
+      - checkout
+      - run:
+          name: Install rustfmt
+          command: rustup component add rustfmt
+      - run:
+          name: Print version information
+          command: cargo fmt -- --version
+      - run:
+          name: Check rustfmt
+          command: cargo fmt -- --check
+  build_debug:
     docker:
       - image: rust:latest
     working_directory: /mnt/crate
     steps:
       - checkout
       - run: git submodule sync
       - run: git submodule update --init
+      - restore_cache:
+          key: cargo-cache-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
+      - run:
+          name: Print version information
+          command: rustc --version; cargo --version
+      - run:
+          name: Build
+          command: cargo build --frozen
+      - persist_to_workspace:
+          root: target
+          paths:
+            - debug/*
```

Comment:
Consider pruning it to only persisting the cached sources; the test will rebuild the binaries anyway.

Reply:
The binaries might be reused as well, at least accordingly to my quick comparison of the build directory between the steps.